src/plugins/management/public/plugin.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { CoreSetup, CoreStart, Plugin } from 'kibana/public';
+import { ManagementStart } from './types';
+// @ts-ignore
+import { management } from './legacy';
+
+export class ManagementPlugin implements Plugin<{}, ManagementStart> {
+  public setup(core: CoreSetup) {
+    return {};
+  }
+
+  public start(core: CoreStart) {
+    return {
+      legacy: management(core.application.capabilities),
```

Comment:
We're passing `core.application.capabilities` here, but it doesn't look like we are doing anything with it in the `management()` fn.

Maybe instead of making the changes to `visibleItems` that were added to `ManagementSection`, we should instead keep the implementation as you've done here, and then in the management function pass the `capabilities` to the `ManagementSection` constructor?

That way `visibleItems` (and therefore the legacy API) remains unchanged.

If we keep it this way, we may need to document the changes to `visibleItems` to be safe